sycl/source/detail/global_handler.cpp

```diff
@@ -66,6 +66,44 @@ std::atomic_uint ObjectUsageCounter::MCounter{0};
 GlobalHandler::GlobalHandler() = default;
 GlobalHandler::~GlobalHandler() = default;
 
+void GlobalHandler::InitXPTIStuff() {
+  // Let subscribers know a new stream is being initialized
+  getXPTIRegistry().initializeStream(SYCL_SYCLCALL_STREAM_NAME, GMajVer,
+                                     GMinVer, GVerStr);
+  xpti::payload_t SYCLPayload("SYCL Interface Layer");
+  uint64_t SYCLInstanceNo;
+  GSYCLCallEvent =
+      xptiMakeEvent("SYCL API Layer", &SYCLPayload, xpti::trace_algorithm_event,
+                    xpti_at::active, &SYCLInstanceNo);
+}
+
+bool hasValue(const char *Value) { return Value && Value[0] != '\0'; }
+
+void GlobalHandler::TraceEventXPTI(const char *Message) {
+  if (!Message)
+    return;
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  if (xptiTraceEnabled()) {
+    uint64_t Uid = xptiGetUniqueId(); // Gets the UID from TLS
+    uint8_t StreamID = xptiRegisterStream(SYCL_SYCLCALL_STREAM_NAME);
```

Comment:
I think diagnostics from exceptions should always be available for any tool.  Most subscribers will always subscribe to "sycl" stream and it makes sense for this stream to be used. Using "sycl.debug" stream for this maybe okay too. Creating too many new streams will create a problem for subscribers that already exist and are subscribing from the "sycl" stream or streams that existed when they created a subscriber. They have to refactor their subscriber to get this data. By using "sycl" we can avoid that and they will continue to get new data when it's available.

Reply:
sorry, didn't catch the point. what exactly requires improvement here?

Reply:
fixed